src/arcam/fmj/state.py

```diff
@@ -255,16 +257,27 @@ def get_source(self) -> Optional[SourceCodes]:
         value = self._state.get(CommandCodes.CURRENT_SOURCE)
         if value is None:
             return None
-        return SourceCodes.from_int(
+        correct_enum = SourceCodes
+        if self._api_model in SOURCE_WRITE_SUPPORTED:
+            correct_enum = DIRECT_SOURCE_CODES.get((self._api_model, self._zn))
+        return correct_enum.from_int(
```

Comment:
I want all system to use same sourcecode lookups. The bytes in the enum for the old codes should move to the new direct source code lookup to.

Reply:
You are missing the point. The old enum byte values 
are wrong for this series as you know.

So i dont want the enum to have any byte codes at all. All series should lookup through the new map added. And it should be renamed to SOURCE_CODES.

Reply:
my understanding of this means we define a map SOURCE_CODES like below

```py
SOURCE_CODES = {
    (ApiModel.API450_SERIES, 1): {
        # mapping source to non rc5 value, 17 entries
    },
    (ApiModel.API450_SERIES, 2): {
        # mapping source to non rc5 value, 17 entries
    },
    (ApiModel.API860_SERIES, 1): {
        # mapping source to non rc5 value, 17 entries
    },
    (ApiModel.API860_SERIES, 2): {
        # mapping source to non rc5 value, 17 entries
    },
    (ApiModel.APIHDA_SERIES, 1): {
        # mapping source to non rc5 value, 17 entries
    },
    (ApiModel.APIHDA_SERIES, 2): {
        # mapping source to non rc5 value, 17 entries
    },
    (ApiModel.APISA_SERIES, 1): {
        # mapping source to non rc5 value, 12 entries
    },
    (ApiModel.APISA_SERIES, 2): {
        # mapping source to non rc5 value, 12 entries
    },
}
```

if the above is correct, a lookup for each of the series would be quite redundant, where we have 6 blocks of code thats essentially copied and pasted.

could we have every entry in the map but the SA series be empty? subsequent lookups for a series with an empty entry in `SOURCE_CODES` would return the default, existing`SourceCodes` enum. i could add a helper function similar to `get_rc5code` to handle this

Reply:
You can define a common variable that is reused for each series in the init of the dict where they are equal.

Reply:
Yup this looks good

Reply:
added tests, was hoping i could get review/approval so we can close the homeassistant issue